spotbugs-tests/src/test/java/edu/umd/cs/findbugs/AbstractIntegrationTest.java

```diff
@@ -142,4 +147,133 @@ protected void performAnalysis(@SlashedClassName final String... analyzeMe) {
                 .toArray(Path[]::new);
         bugReporter = runner.run(paths);
     }
+
+    final protected void assertNoBugType(String bugType) {
+        assertBugTypeCount(bugType, 0);
+    }
+
+    final protected void assertBugTypeCount(String bugType, int count) {
+        final BugInstanceMatcher matcher = new BugInstanceMatcherBuilder()
+                .bugType(bugType)
+                .build();
+        assertThat(getBugCollection(), containsExactly(count, matcher));
+    }
+
+    final protected void assertNoBugInClass(String bugType, String className) {
```

Comment:
Thanks a lot for the improvement!
Unless I misunderstood the `className` argument here is the "simple name", not the full (dotted or slashed) qualified name, maybe there should be a javadoc comment because otherwise that might be confusing

Reply:
It depends on the implementation of BugInstanceMatcher, which matches any of the followings: 
- dotted class name, 
- the simple name,
- in case of inner classes the simple name with the container class separated by `$`.

See the relevant codepart:
https://github.com/spotbugs/spotbugs/blob/c4b3fa473d6df297539aa20c57770b90cb9aff52/test-harness/src/main/java/edu/umd/cs/findbugs/test/matcher/BugInstanceMatcher.java#L103-L115

It looks like to me, only `className` has several options or may be ambiguous. I didn't want to write in one class's javadoc something (several times) depending on another's implementation, so updated `BugInstanceMatcher`'s constructor's javadoc and refered that in the javadoc of `AbstactIntegrationTest`. Unfortunately, I couldn't find any way to refer in the javadoc another method's parameter, or another class's private field, so this was the best I could come up with. LMK if you have a better idea.

Reply:
Thanks for looking into it and the clarification, this looks good to me!